nntrainer/models/neuralnet.cpp

```diff
@@ -556,9 +556,42 @@ sharedConstTensors NeuralNetwork::inference(sharedConstTensors X,
   return out;
 }
 
-std::vector<float *> NeuralNetwork::inference(std::vector<float *> &input,
-                                              unsigned int batch_size) {
-  sharedConstTensors input_tensors;
+sharedConstTensors NeuralNetwork::inference(sharedConstTensors X,
+                                            sharedConstTensors label,
+                                            bool free_mem) {
+  if (model_graph.getBatchSize() != X[0]->batch()) {
+    model_graph.setBatchSize(X[0]->batch());
+  }
+
+  sharedConstTensors out;
+  if (!validateInput(X))
+    throw std::invalid_argument("Input validation failed.");
+
+  allocate(ExecutionMode::INFERENCE);
+
+  START_PROFILE(profile::NN_FORWARD);
+  out = forwarding(X, label, false);
+  END_PROFILE(profile::NN_FORWARD);
+
+  if (free_mem)
+    /**
+     * Free the memory needed for training before exiting.
+     * Note that this does not free the weights for the model.
+     * Weights of the model will be freed when the model is destroyed.
+     */
+    model_graph.deallocateTensors(false);
+
+  /** Clear the set inputs and labels */
+  setLabels({}, model_graph.getLabelList());
+  setInputs({}, model_graph.getInputList());
+
+  return out;
+}
+
+std::vector<float *> NeuralNetwork::inference(unsigned int batch_size,
+                                              std::vector<float *> &input,
+                                              std::vector<float *> &label) {
```

Comment:
fyi) if we change the signature here, nnstreamer nntrainer filter has to change as well.

Reply:
Lets set `label = {}` for default. With https://github.com/nnstreamer/nntrainer/pull/1601#discussion_r721831285, this can be supported.